deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml

```diff
@@ -110,6 +110,20 @@ resource/remove_k8s_pod_pod_name:
     - action: delete
       key: k8s.pod.pod_name
 
+transform/prepare_routing:
+  error_mode: ignore
+  metric_statements:
+    - context: metric
+      statements:
+        - set(resource.attributes["http_listener_v2_path"], "/prometheus.metrics.apiserver") where resource.attributes["job"] == "apiserver"
+        - set(resource.attributes["http_listener_v2_path"], "/prometheus.metrics.control-plane.coredns") where resource.attributes["job"] == "coredns"
+        - set(resource.attributes["http_listener_v2_path"], "/prometheus.metrics.control-plane.kube-etcd") where resource.attributes["job"] == "kube-etcd"
+        - set(resource.attributes["http_listener_v2_path"], "/prometheus.metrics.controller-manager") where resource.attributes["job"] == "kubelet" and IsMatch(name, "^cloudprovider_.*")
+        - set(resource.attributes["http_listener_v2_path"], "/prometheus.metrics.kubelet") where resource.attributes["job"] == "kubelet" and not IsMatch(name, "^cloudprovider_.*")
+        - set(resource.attributes["http_listener_v2_path"], "/prometheus.metrics.node") where resource.attributes["job"] == "node-exporter"
+        - set(resource.attributes["http_listener_v2_path"], "/prometheus.metrics.scheduler") where resource.attributes["job"] == "kube-scheduler"
+        - set(resource.attributes["http_listener_v2_path"], "/prometheus.metrics.state") where resource.attributes["job"] == "kube-state-metrics"
```

Comment:
Why not just put these conditions in the routing processor directly?

Reply:
Documentation says that routing processor works only with resource attributes, but I need to use metric name in two places :cry:

Reply:
Do we actually need to do that? Why do the `cloudprovider_.*` metrics need to be routed to the controller-manager source anyway?

Reply:
We do not, but it will be breaking change if someone is using `sourceName` in their dashboards

Reply:
Are you sure this is even correct? According to https://kubernetes.io/docs/concepts/cluster-administration/system-metrics/#kube-controller-manager-metrics, these are in fact controller-manager metrics. Looks like what we have right now doesn't work at all.

Reply:
Need to revalidate with current state, but according to this PR: https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/56
the `kubelet` is in purpose

Reply:
@swiatekm-sumo 

![image](https://github.com/SumoLogic/sumologic-kubernetes-collection/assets/58699848/af57f2e5-8171-4705-a856-edd37a82e5ac)

Reply:
Allright, let's do it this way then. But we do need to make sure we're setting this correctly at the resource level.